packages/x-data-grid/src/material/index.tsx

```diff
@@ -103,4 +106,18 @@ const materialSlots: GridBaseSlots & GridIconSlotsComponent = {
   baseChip: MUIChip,
 };
 
+const materialSlots: GridBaseSlots & GridIconSlotsComponent = {
+  ...baseSlots,
+  ...iconSlots,
+};
+
 export default materialSlots;
+
+function BaseMenuItem(props: GridSlotProps['baseMenuItem']) {
+  const { iconStart, iconEnd, children, ...other } = props;
+  return React.createElement(MUIMenuItem, other, [
```

Comment:
Why not use JSX? To avoid `{...other}`?

Reply:
Yes, it's unfortunate that JSX doesn't support passing props as-is without destructuring into a new object.